airflow/providers/google/cloud/hooks/dataplex.py

```diff
@@ -361,3 +374,434 @@ def get_lake(
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+        return result
+
+    def wait_for_data_scan_job(
+        self,
+        data_scan_id: str,
+        fail_on_dq_failure: bool,
+        job_id: str | None = None,
+        fail_on_job_failure: bool = True,
+        project_id: str | None = None,
+        region: str | None = None,
+        wait_time: int = 10,
+        timeout: float | None = None,
+        fail_on_timeout: bool = True,
+    ) -> None:
+        """
+        Wait for Dataplex data scan job.
+
+        :param job_id: Required. The job_id to wait for.
+        :param data_scan_id: Required. Data Quality scan identifier.
+        :param region: Required. The ID of the Google Cloud region that the lake belongs to.
+        :param project_id: Optional. Google Cloud project ID.
+        :param wait_time: Number of seconds between checks.
+        :param timeout: The amount of time, in seconds, to wait for an execution job the result.
+        :param fail_on_dq_failure: If set to true and not all Data Quality scan rules have been passed,
+            an exception is thrown. If set to false and not all Data Quality scan rules have been passed,
+            execution will finish with success.
+        :param fail_on_job_failure: If set to true and job fails, an exception is thrown.
+        :param fail_on_timeout: If set to true and timeout, an exception is thrown.
+        """
+        start = time.monotonic()
+        state = None
+        while state not in (
+            DataScanJob.State.CANCELLED,
+            DataScanJob.State.FAILED,
+            DataScanJob.State.SUCCEEDED,
+        ):
+            if timeout and start + timeout < time.monotonic():
+                if fail_on_timeout:
+                    raise AirflowException(
+                        f"Timeout: data quality scan {job_id} is not ready after {timeout}s"
+                    )
+                break
+            time.sleep(wait_time)
+            try:
+                job = self.get_data_scan_job(
+                    job_id=job_id,
+                    data_scan_id=data_scan_id,
+                    project_id=project_id,
+                    region=region,
+                )
+                state = job.state
+            except Exception as err:
+                self.log.info("Retrying. Dataplex API returned error when waiting for job: %s", err)
+
+            if fail_on_dq_failure:
+                if state == DataScanJob.State.SUCCEEDED and not job.data_quality_result.passed:
+                    raise AirflowException(f"Data Quality scan failed: {data_scan_id}")
+            if state == DataScanJob.State.FAILED:
+                if fail_on_job_failure:
+                    raise AirflowException(f"Job failed:\n{job_id}")
+                break
+            if state == DataScanJob.State.CANCELLED:
+                raise AirflowException(f"Job was cancelled:\n{job_id}")
```

Comment:
```suggestion
        start = time.monotonic()
        state = None
        while state not in (
            DataScanJob.State.CANCELLED,
            DataScanJob.State.FAILED,
            DataScanJob.State.SUCCEEDED,
        ):
            if execution_timeout and start + execution_timeout < time.monotonic():
                raise AirflowException(
                        f"Timeout: data quality scan {job_id} is not ready after {timeout}s"
                    )
            time.sleep(wait_time)
            try:
                job = self.get_data_scan_job(
                    job_id=job_id,
                    data_scan_id=data_scan_id,
                    project_id=project_id,
                    region=region,
                )
                state = job.state
            except Exception as err:
                self.log.info("Retrying. Dataplex API returned error when waiting for job: %s", err)
       return state
           
```

Reply:
changed, but here we have to return the job, not just the status

Reply:
got it, thanks!